test/gas/gas.js

```diff
@@ -0,0 +1,563 @@
+const assert = require("assert");
+const sleep = require("../helpers/utils/sleep");
+const bootstrap = require("../helpers/contract/bootstrap");
+const getWeb3 = require("../helpers/web3/getWeb3");
+const isGasExpenseCorrect = require("./lib/isGasExpenseCorrect");
+const testTransactionEstimate = require("./lib/transactionEstimate");
+const toBytes = require("./lib/toBytes");
+const deployContract = require("./lib/customContractDeploy");
+
+const RSCLEAR_REFUND = 15000;
+const RSCLEAR_REFUND_FOR_RESETTING_DIRTY_SLOT_TO_ZERO = 19800;
+const RSELFDESTRUCT_REFUND = 24000;
+
+// Thanks solc. At least this works!
+// This removes solc's overzealous uncaughtException event handler.
+process.removeAllListeners("uncaughtException");
+
+describe("Gas", () => {
+  const mainContract = "EstimateGas";
+  const contractFilenames = [];
+  const contractPath = "../../contracts/gas/";
+
+  const mnemonic = "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat";
+  const options = { mnemonic };
+  const services = bootstrap(mainContract, contractFilenames, options, contractPath);
+
+  let deploymentReceipt;
+
+  before("Adding funds to the contract", async function() {
+    this.timeout(10000);
+    const { accounts, bytecode, contract } = services;
+
+    const promiEvent = contract.deploy({ data: bytecode }).send({
+      from: accounts[0],
+      gas: 3141592
+    });
+
+    promiEvent.on("receipt", (receipt) => {
+      deploymentReceipt = receipt;
+    });
+  });
+
+  describe("Refunds", () => {
+    it(
+      "accounts for Rsclear Refund in gasEstimate when a dirty storage slot is reset and it's original " +
+        " value is 0",
+      async() => {
+        const { accounts, instance, provider } = services;
+        const from = accounts[0];
+        const options = { from, gas: 5000000 };
+
+        // prime storage by making sure it is set to 0
+        await instance.methods.reset().send(options);
+
+        // update storage and then reset it back to 0
+        const method = instance.methods.triggerRsclearRefund();
+
+        const gasEstimate = await method.estimateGas(options);
+
+        const { gasUsed, cumulativeGasUsed } = await method.send({ from, gas: gasEstimate });
```

Comment:
No need to change this now, but for future refactoring let's avoid switching to destructuring syntax here (we used to keep a reference to the entire receipt before).

On the one hand destructuring is nice because there's less to type later on down the line, but on the other it squashes  everything into the local namespace/scope. I think keeping things namespaced (e.g. `receipt.gasUsed` instead of `gasUsed`) makes it easier to understand the source of the data in question, and makes it less likely to have naming collisions as logic is added over time.

Reply:
That said, I also wouldn't change things that are already in a destructured style to go back to the object reference style. There isn't enough benefit either way to make it worth your time

Reply:
ok, will keep this in mind.

Reply:
Reverted back to `receipts`.